scalableminds/webknossos
# Add functions to get and set segment colors to the frontend API

Also, fix a bug in the segment color picker where the color that was preselected after opening the picker was wrong.
The `hexToRgb` utils function returned the color in [0, 255], but later it was assumed to be in [0, 1] and was scaled to 255 again. Because `hexToRgb` was only used for the segment color picker, I changed its return value to be in [0, 1] which we usually use in other related functions.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Create an annotation on a dataset with a segmentation layer. Select a segment and compute an ad-hoc mesh for it.
- Change its color using the segments list.
  - Reopen the color picker. The preselected color should be correct.
- Use `webknossos.apiReady(3).then(async (api) => {console.log(api.data.getSegmentColor(<YOUR_SEGMENT_ID>))})` to get the color of the segment via the api -> The color should be the previously selected color in RGB in the interval [0, 1]
- Use `webknossos.apiReady(3).then(async (api) => {console.log(api.data.setSegmentColor(<YOUR_SEGMENT_ID>, [0, 1, 1]))})` to set the color of the segment (and mesh) -> The segment and mesh should change their color to turquoise

### Issues:
- contributes to #6696 

------
(Please delete unneeded items, merge only when none are left open)
